src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/BaseRunner.cs

```diff
@@ -208,19 +208,31 @@ private async Task<(TestResult, string)> RunAsync2(TestCase testCase)
                     //    Errors: Error 15-16: Incompatible types for comparison. These types can't be compared: UntypedObject, UntypedObject.
                     var expectedCompilerError = expected.StartsWith("Errors: Error") || expected.StartsWith("Errors: Warning"); // $$$ Match error message. 
                     if (expectedCompilerError)
-                    {
-                        var msg = $"Errors: " + string.Join("\r\n", runResult.Errors.Select(err => err.ToString()).ToArray());
+                    {
+                        string[] expectedStrArr = expected.Replace("Errors: ", string.Empty).Split("|");
+                        string[] actualStrArr = runResult.Errors.Select(err => err.ToString()).ToArray();
+
+                        bool isValid = true;
+                        foreach (var exp in expectedStrArr)
+                        {
+                            if (!actualStrArr.Contains(exp))
+                            {
+                                isValid = false;
+                                break;
+                            }
+                        }
+
+                        var msg = $"Errors: " + string.Join("\r\n", actualStrArr);
                         var actualStr = msg.Replace("\r\n", "|").Replace("\n", "|");
 
                         // Try both unaltered comparison and by replacing Decimal with Number for errors,
                         // for tests that are run with and without NumberIsFloat set.
-                        if (actualStr.Contains(expected))
+                        if (isValid)
                         {
                             // Compiler errors result in exceptions
                             return (TestResult.Pass, null);
                         }
-                        else if (NumberIsFloat && expected.StartsWith("Errors:") &&
-                                 actualStr.Contains(Regex.Replace(expected, "(?<!Number,)(\\s|'|\\()Decimal(\\s|'|,|\\.|\\))", "$1Number$2")))
+                        else if (NumberIsFloat && actualStr.Contains(Regex.Replace(expected, "(?<!Number,)(\\s|'|\\()Decimal(\\s|'|,|\\.|\\))", "$1Number$2")))

```

Comment:
>NumberIsFloat && actualStr.Contains(Regex.Replace(expected, "(?<!Number,)(\\s|'|\\()Decimal(\\s|'|,|\\.|\\))", "$1Number$2") [](http://example.com/codeflow?start=33&length=124)

We should move this check to inside the loop (line 218) - this way it will work with out-of-order messages as well

Reply:
updated.